File: src/pt/up/fe/luisfonseca/cp/api/ResponseHandler.java
package pt.up.fe.luisfonseca.cp.api;

public interface ResponseHandler<T> {

	enum ERROR_TYPE{
		CANCELLED,
		AUTHENTICATION,
		NETWORK,
		GENERAL
	};
	public void onError( ERROR_TYPE error );
	
	public void onResultReceived( T results );
}

File: src/pt/up/fe/luisfonseca/cp/api/ResponseHandler.scala
package pt.up.fe.luisfonseca.cp.api

object ResponseHandler {
  object Error extends Enumeration {
    type Type = Value
    val Cancelled,
		Authentication,
		Network,
		General = Value
  }
}

trait ResponseHandler[T] {
	
  def onError(error: ResponseHandler.Error.Type)
  
  def onResultReceived(result: T)
}